grpc/server/src/main/java/io/helidon/grpc/server/ProtoReflectionService.java

```diff
@@ -594,4 +596,15 @@ private void processExtension(FieldDescriptor extension, FileDescriptor fd) {
             fileDescriptorsByExtensionAndNumber.get(extensionName).put(extensionNumber, fd);
         }
     }
+
+    private <T> T accessIndex(Supplier<T> operation) {
+        try {
+            indexAccess.acquire();
+            T result = operation.get();
+            indexAccess.release();
```

Comment:
The release will not occur if `operation.get()` fails.

Reply:
Fixed.